js/modal1.js

```diff
@@ -0,0 +1,107 @@
+import { fetchAvailableTimes } from "./apiService.js";
+
+console.log(document.querySelector('.booking__searchButton'));
+console.log(document.querySelector('.custom__date'));
+/*Ronjas modal script*/ 
+console.log("hello from modal 1"); //To see if it works
+
+const modal = document.querySelector('.booking-modal');
+const steps = document.querySelectorAll('.booking-modal__step')
+const step1 = document.querySelector('#step1');
+const step2 = document.querySelector('#step2');
+const step3 = document.querySelector('#step3');
+
+console.log(modal);
+
+export const handleBookButtonClick = ()=>{
+console.log('Hello from the book btn');
+  
+modal.style.display = 'block';
+step2.style.display = 'none';
+step3.style.display = 'none';
+};
+
+function showSteps(stepNumber){
+    modal.style.display='block';
+    steps.forEach((step, index) => {
+        step.style.display = index + 1 === stepNumber ? 'block' : 'none';
+        step.classList.toggle('booking-modal__step--active', index + 1 === stepNumber);
+    });
+}
+
+document.addEventListener('DOMContentLoaded', () => {
+    modal.style.display = 'none'; 
+    showSteps(1); //step 1 shows
+})
+
+const searchButton = document.querySelector('.booking__search-btn');
+searchButton.addEventListener('click', (e) => {
+    e.preventDefault();
+    showSteps(2); //Showing step 2 of modal
```

Comment:
your searchButton is currently included within a <form>, causing the browser's default form submission behavior to trigger when a date is selected and clicked enter. This results in an automatic jump to step 2 without giving the user a chance to view the available slots. To avoid this, you could use a different tag like <div> or <section>instead of a <form>. This would prevent the default form submission behavior and give you full control over the interaction flow. :)